frontend/src/Components/Game.jsx

```diff
@@ -157,85 +156,133 @@ class Game extends React.Component {
         this.setState({ buttonPressed: gamepadState });
       }
       window.requestAnimationFrame(update);
+
+      if (this.state.canMove && this.state.gampadConnected) {
+        this.setState({ canMove: false });
+        setTimeout(() => {
+          this.setState({ canMove: true });
+        }, 120);
+      }
     };
     window.requestAnimationFrame(update);
+  }
 
-   /* Player  Movement  */
-  makeNpcMove = setInterval(() => {
-    if (this.state.NPC.isAlive) {
-      this.pathFinding(
-        this.state.NPC.x,
-        this.state.NPC.y,
-        this.state.x,
-        this.state.y
-      );
-    } else {
-      clearInterval(this.makeNpcMove);
-    }
-  }, 300);
-
+  /* Player  Movement  */
   gamepadMove() {
     let newPosition;
     let x = this.state.x;
     let y = this.state.y;
-    let newDirection = "down";
-    if (this.state.buttonPressed.axes[1] === "1.00") {
+    let newDirection;
+
+    if (
+      this.state.buttonPressed.axes[1] === "1.00" &&
+      this.state.direction === "down"
+    ) {
+      this.setState({ pressKey: this.state.pressKey + 1 });
       newPosition = y + 1;
       newDirection = "down";
       if (this.isMovePossible(x, y + 1)) {
+        this.setState({ pressKey: this.state.pressKey + 1 });
         this.setState({
           direction: newDirection,
           y: newPosition,
           keyName: "ArrowDown"
         });
       } else {
-        this.playBounce();
+        this.props.playBounce();
       }
-    } else if (this.state.buttonPressed.axes[1] === "-1.00") {
+    } else if (
+      this.state.buttonPressed.axes[1] === "-1.00" &&
+      this.state.direction === "up"
+    ) {
+      this.setState({ pressKey: this.state.pressKey + 1 });
       newPosition = y - 1;
       newDirection = "up";
       if (this.isMovePossible(x, y - 1)) {
+        this.setState({ pressKey: this.state.pressKey + 1 });
```

Comment:
Why do a separate `setState` for pressKey ? It'd be better to update it in the same as below.
(Same comment for several occurences in this file)